src/lavatory/utils/artifactory.py

```diff
@@ -191,6 +160,30 @@ def get_all_repo_artifacts(self, depth=None, item_type='file', with_properties=T
         artifacts = self.filter(item_type=item_type, depth=depth, fields=fields)
         return sorted(artifacts, key=lambda k: k['path'])
 
+    def time_based_retention(self, keep_days=None, item_type='file', extra_aql=[]):
+        """Retains artifacts based on number of days since creation.
+
+            extra_aql example: [{"@deployed": {"$match": "dev"}}, {"@deployed": {"$nmatch": "prod"}}]
+            This would search for artifacts that were created after <keep_days> with
+            property "deployed" equal to dev and not equal to prod.
+
+        Args:
+            keep_days (int): Number of days to keep an artifact.
+            item_type (str): The item type to search for (file/folder/any). 
+            extra_aql (list). List of extra AQL terms to apply to search
+
+        
+        Return:
+            list: List of artifacts matching retention policy
+        """
+        now = datetime.datetime.now()
+        before = now - datetime.timedelta(days=keep_days)
+        created_before = before.strftime("%Y-%m-%dT%H:%M:%SZ")
+        aql_terms = [{"created": {"$lt": created_before}}]
+        aql_terms.extend(extra_aql)
+        purgable_artifacts = self.filter(item_type=item_type, depth=None, terms=aql_terms)
```

Comment:
It works with none:

```        if depth:
            terms.append({"depth": {"$eq": depth}})```

If you are looking at artifacts (as opposed to folders) you do not need a depth, it will just go as deep as it needs